File: backend/debug_clientes.js

const { Pool } = require('pg');
require('dotenv').config();

const pool = new Pool({
    host: process.env.DB_HOST || 'localhost',
    port: process.env.DB_PORT || 5432,
    database: process.env.DB_NAME || 'basesales',
    user: process.env.DB_USER || 'postgres',
    password: process.env.DB_PASSWORD || '@12Pilabo',
});

async function check() {
    try {
        const res = await pool.query("SELECT * FROM information_schema.tables WHERE table_name = 'clientes'");
        console.log("Table 'clientes' exists:", res.rows.length > 0);

        if (res.rows.length > 0) {
            const cols = await pool.query("SELECT column_name FROM information_schema.columns WHERE table_name = 'clientes' ORDER BY column_name");
            console.log("Columns:", cols.rows.map(r => r.column_name));
        }
    } catch (e) {
        console.error(e);
    } finally {
        pool.end();
    }
}

check();

File: backend/debug_clientes.py
import psycopg2
from psycopg2.extras import RealDictCursor

conn_params = {
    "host": "node254557-salesmaster.sp1.br.saveincloud.net.br",
    "port": 13062,
    "database": "basesales",
    "user": "webadmin",
    "password": "ytAyO0u043"
}

def analyze_clientes(schema):
    try:
        conn = psycopg2.connect(**conn_params)
        cur = conn.cursor(cursor_factory=RealDictCursor)
        
        print(f"\n--- Analyzing {schema}.clientes ---")
        
        # Check column names
        cur.execute(f"SELECT column_name FROM information_schema.columns WHERE table_schema = '{schema}' AND table_name = 'clientes'")
        cols = [r['column_name'] for r in cur.fetchall()]
        print(f"Columns: {cols}")
        
        # Check counts of cli_tipopes
        if 'cli_tipopes' in cols:
            cur.execute(f"SELECT cli_tipopes, count(*) as total FROM {schema}.clientes GROUP BY cli_tipopes")
            print(f"cli_tipopes counts: {cur.fetchall()}")
            
        # Check counts of cli_ativo
        if 'cli_ativo' in cols:
            cur.execute(f"SELECT cli_ativo, count(*) as total FROM {schema}.clientes GROUP BY cli_ativo")
            print(f"cli_ativo counts: {cur.fetchall()}")

        # Check counts of cli_status
        if 'cli_status' in cols:
            cur.execute(f"SELECT cli_status, count(*) as total FROM {schema}.clientes GROUP BY cli_status")
            print(f"cli_status counts: {cur.fetchall()}")

        cur.close()
        conn.close()
    except Exception as e:
        print(f"Error in {schema}: {e}")

if __name__ == "__main__":
    analyze_clientes('ro_consult')
    analyze_clientes('barrosrep')
